tests/aggregation/tests.py

```diff
@@ -2359,3 +2360,86 @@ def test_aggregate_combined_queries(self):
             )
         )
         self.assertEqual(qs.count(), 3)
+
+
+class JSONArrayAggTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.a1 = Author.objects.create(name="Adrian Holovaty", age=34)
+        cls.a2 = Author.objects.create(name="Jacob Kaplan-Moss", age=35)
+        cls.a3 = Author.objects.create(name="Brad Dayley", age=45)
+
+        cls.p1 = Publisher.objects.create(
+            name="Apress", num_awards=3, duration=datetime.timedelta(days=1)
+        )
+
+        cls.b1 = Book.objects.create(
+            isbn="159059725",
+            name="The Definitive Guide to Django: Web Development Done Right",
+            pages=447,
+            rating=4.5,
+            price=Decimal("30.00"),
+            contact=cls.a1,
+            publisher=cls.p1,
+            pubdate=datetime.date(2007, 12, 6),
+        )
+
+    def test_JSONArrayAgg(self):
+        vals = Author.objects.aggregate(jsonarrayagg=JSONArrayAgg("name"))
```

Comment:
It'd be great if the tests covered relationships such as `books__name` as the way the tests are currently designed they never result in JOINs!